### PR TITLE
Document.createAttributeNS

### PR DESCRIPTION
**Pull request by edemaine**
#### Summary
* Adds documentation for the `Document.createAttributeNS` DOM interface (and "see also" links)
* Small grammar fix in [`Document.createAttribute`](https://developer.mozilla.org/en-US/docs/Web/API/Document/createAttribute)
* Small improvements to [`Document.createElementNS`](https://developer.mozilla.org/en-US/docs/Web/API/Document/createElementNS): fix broken link (I hope) and add possible exceptions.

#### Motivation
Though I guess rarely used over [`Element.setAttributeNS`](https://developer.mozilla.org/en-US/docs/Web/API/Element/setAttributeNS), this seems to be a core part of the DOM spec, [defined with `createAttribute`](https://dom.spec.whatwg.org/#dom-document-createattribute).

#### Supporting details
Based on the existing documentation for [`Document.createAttribute`](https://developer.mozilla.org/en-US/docs/Web/API/Document/createAttribute) and [`Document.createElementNS`](https://developer.mozilla.org/en-US/docs/Web/API/Document/createElementNS).

I've never created a page before, so I'm not sure whether I need to do something to get the Specifications or Browser compatibility to work. Or to get a crosslink back from the DOM spec.

#### Metadata
- [x] Adds a new document
- [ ] Rewrites (or significantly expands) a document
- [x] Fixes a typo, bug, or other error

### CODE REVIEW

File: files/en-us/web/api/document/createattributens/index.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+- [HTML](/en-US/docs/Web/HTML)
+  - : `http://www.w3.org/1999/xhtml`
+- [SVG](/en-US/docs/Web/SVG)
+  - : `http://www.w3.org/2000/svg`
+- [MathML](/en-US/docs/Web/MathML)
+  - : `http://www.w3.org/1998/Math/MathML`
+- [XUL](/en-US/docs/Mozilla/Tech/XUL) {{Non-standard_Inline}}
+  - : `http://www.mozilla.org/keymaster/gatekeeper/there.is.only.xul`
+- [XBL](/en-US/docs/Mozilla/Tech/XBL) {{Non-standard_Inline}} {{Deprecated_Inline}}
+  - : `http://www.mozilla.org/xbl`
```
Comment (teoli2003) [inline files/en-us/web/api/document/createattributens/index.md:57]: XUL and XBL are old technologies that are no more relevant. Let's not cite them:
```suggestion
```

File: files/en-us/web/api/document/createattributens/index.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+## Syntax
+
+```js
+createAttributeNS(namespaceURI, qualifiedName)
+```
+
+### Parameters
+
+- _namespaceURI_
```
Comment (teoli2003) [inline files/en-us/web/api/document/createattributens/index.md:27]:
```suggestion
- `namespaceURI`
```

File: files/en-us/web/api/document/createattributens/index.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+  - : A string that specifies the [namespace URI](https://www.w3.org/TR/2004/REC-DOM-Level-3-Core-20040407/glossary.html#dt-namespaceURI) to associate with the attribute.
+    The {{DOMxRef("attribute.namespaceURI", "namespaceURI")}} property of the created attribute is initialized with the value of _namespaceURI_.
+    See [Valid Namespace URIs](#important_namespace_uris).
+- _qualifiedName_
```
Comment (teoli2003) [inline files/en-us/web/api/document/createattributens/index.md:31]:
```suggestion
- `qualifiedName`
```

File: files/en-us/web/api/document/createattributens/index.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+```js
+createAttributeNS(namespaceURI, qualifiedName)
+```
+
+### Parameters
+
+- _namespaceURI_
+  - : A string that specifies the [namespace URI](https://www.w3.org/TR/2004/REC-DOM-Level-3-Core-20040407/glossary.html#dt-namespaceURI) to associate with the attribute.
+    The {{DOMxRef("attribute.namespaceURI", "namespaceURI")}} property of the created attribute is initialized with the value of _namespaceURI_.
```
Comment (teoli2003) [inline files/en-us/web/api/document/createattributens/index.md:29]:
```suggestion
    The {{DOMxRef("attribute.namespaceURI", "namespaceURI")}} property of the created attribute is initialized with the value of `namespaceURI`.
```

File: files/en-us/web/api/document/createattributens/index.md [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+    See [Valid Namespace URIs](#important_namespace_uris).
+- _qualifiedName_
+  - : A string that specifies the name of attribute to be created.
+    The {{DOMxRef("attribute.name", "name")}} property of the created attribute is initialized with the value of _qualifiedName_.
```
Comment (teoli2003) [inline files/en-us/web/api/document/createattributens/index.md:33]:
```suggestion
    The {{DOMxRef("attribute.name", "name")}} property of the created attribute is initialized with the value of `qualifiedName`.
```

**Review by teoli2003 — CHANGES_REQUESTED** [review]
Thanks a lot!

This is mostly good. I removed XUL and XBL mentions as these are dead technologies and used the modern way of writing parameters (on the new page, not on the modified ones, though we will have to update them at some point).

**Review by teoli2003 — APPROVED** [review]
👍

**Comment by teoli2003** [comment]
Thanks a lot! I'm always glad to see missing pages being written.

🎉

**Comment by edemaine** [comment]
Thanks for the comments!

> not on the modified ones, though we will have to update them at some point

Would you like me to edit the other page in this PR, or leave it for someone else? Oops, given the merge, I guess I'll leave that for others.

Should I be worried about "No specification data found for api.Document.createAttributeNS."?

**Comment by teoli2003** [comment]
This means the mdn/browser-compat-data is missing the spec_url field for `api.Document.createAttributeNS`. We should add it. Do you want to do the PR? (If not, I'll do it, no worry).

(Then we will have to wait for the next release of browser-compat-data, normally next week. Then the content here will appear correctly)

**Comment by edemaine** [comment]
I started looking at https://github.com/mdn/browser-compat-data/blob/aaed37e15d43fff035c6e1fc9fff2fff4b590d37/api/Document.json but it's a fair amount to grok (and I'm not sure how to know what browser versions they were added...), so feel free to make the change without me.  Thanks!

**Comment by teoli2003** [comment]
I'll do it. And cc/ you there so you can see what was missing!

Thanks again!

**Comment by teoli2003** [comment]
Sorry, I'm tired: I already did it this morning and forgot ( https://github.com/mdn/browser-compat-data/pull/16082 ). It already got merged, so it will be published this week or next week. We just need to be patient. :-)

**Pull request merged**